contracts/RenPool.sol

```diff
@@ -70,16 +77,44 @@ contract RenPool {
         uint senderBalance = balances[sender];
 
         require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
-        require(isLocked == false, "Pool is locked");
+        if(!isLocked){
+            totalPooled -= _amount;
+            balances[sender] -= _amount;
+            renToken.transfer(sender, _amount);
+            
+            emit RenWithdrawal(sender, _amount);
+
+        }
+        else{
+            // Pool is locked, withdraw will be put in the queue
+            withdrawRequests.push(WithdrawRequest(msg.sender, _amount));           
+        }
 
-        renToken.transfer(sender, _amount);
-        totalPooled -= _amount;
-        balances[sender] -= _amount;
 
-        emit RenWithdrawal(sender, _amount);
     }
 
     function balanceOf(address _addr) external view returns(uint) {
         return balances[_addr];
     }
+
+    function fullfillWithdrawRequest(uint _withdrawId) external payable{
+        // If pool is locked, look if there is a withdraw queue
```

Comment:
instead of _withdrawId I'd call this _index. It's the array's index isn't it? Also, I think payable only applies to ETH payment, no need for ERC20 tokens

Reply:
Yes, you're right. And well spot for the payable, it was this way before implementing the ERC20 but I forgot to delete it